iceoryx_posh/test/integrationtests/test_service_discovery.cpp

```diff
@@ -473,31 +458,38 @@ TEST_F(ServiceDiscovery_test, FindServiceReturnsMaxServices)
 
     auto serviceContainer = sut.findService(SERVICE, iox::capro::Wildcard, iox::capro::Wildcard);
 
-    ASSERT_FALSE(serviceContainer.has_error());
-    EXPECT_THAT(serviceContainer.value().size(), Eq(iox::MAX_NUMBER_OF_SERVICES));
-    EXPECT_TRUE(serviceContainer.value() == serviceContainerExp);
+
+    EXPECT_THAT(serviceContainer.size(), Eq(iox::MAX_NUMBER_OF_SERVICES));
+    EXPECT_TRUE(serviceContainer == serviceContainerExp);
 }
 
-/// @todo #415 #1074 this test is affected by the limits we set for service discovery,
-/// if the container capacity is larger than what we can send with sockets
-/// we will never cause an overflow
-TEST_F(ServiceDiscovery_test, FindServiceReturnsContainerOverflowErrorWhenMoreThanMaxServicesAreFound)
+static void testCallback(ServiceDiscovery* const ServiceDiscoveryPointer)
 {
-    ::testing::Test::RecordProperty("TEST_ID", "f2f8d8c0-8712-4e7a-9e33-2b2a918f8a71");
-    const IdString_t SERVICE = "s";
-    const size_t numberOfInstances = (iox::MAX_NUMBER_OF_SERVICES + 1);
-    iox::cxx::vector<iox::popo::UntypedPublisher, numberOfInstances> publishers;
-    for (size_t i = 0; i < numberOfInstances; i++)
-    {
-        std::string instance = "i" + iox::cxx::convert::toString(i);
-        iox::capro::ServiceDescription SERVICE_DESCRIPTION(
-            SERVICE, IdString_t(iox::cxx::TruncateToCapacity, instance), "foo");
-        publishers.emplace_back(SERVICE_DESCRIPTION);
-    }
+    IOX_DISCARD_RESULT(ServiceDiscoveryPointer);
+}
 
-    auto serviceContainer = sut.findService(SERVICE, iox::capro::Wildcard, iox::capro::Wildcard);
+TEST_F(ServiceDiscovery_test, ServiceDiscoveryIsAttachableToWaitSet)
```

Comment:
@MatthiasKillat This is how the attaching to the `WaitSet` looks like. Under the hood the `popo::Subscriber` is attached. I'll continue writing the notification-based unit test cases, or did you work on this already?

Reply:
@mossmaurice I did not write the unit test cases for this.

Personally I am not sure it is nice to force the user to use the waitset like this, especially as the events are not ideally named (as they are `iox::popo::SubscriberEvent::DATA_RECEIVED` and expose an implementation detail). What we can and do wait for is a change of the registry.

On the other hand we can (I think) attach multiple waitsets, i.e. wait in different threads for change.

I am also wondering whether it is possible to miss changes to the registry, i.e. it is cumbersome to use together with find). This is what I am experimenting with and were I think a dedicated API that is easier to use would be better. In theory we can have both, attaching where the user needs to create her own abstractions for wait or we can already provide easier to use abstractions for wait.

Another reason to have the ability to attach(via the waitset interface) is that it works with the listener as well.

Reply:
Good point with the event name, fixed that. As we discussed today, not event are missed. Let's use your dedicated API as an example.